From febae98bcecd7f4197e196d9cc00f8834d8c4d55 Mon Sep 17 00:00:00 2001
From: "Vishnu Swaroop Kumar Sarma, Duddu" <duddu.swaroop@intel.com>
Date: Fri, 21 Sep 2018 15:54:40 +0530
Subject: [PATCH] UGW_SW-30653: ParserFlagLSB_Mask of TCP Unicast PCE rule is
 changed wronly ( 0xFFF3 -> 0).Switch_api deletes a series of entries
 requested by PPA_PAE via delete_pce_rule, but switch does perform delete in a
 Flag Table due to parser 1 which is 32:63 was not handled properly for
 GRX500, because GRX500 does not have this field.

---
 drivers/net/ethernet/lantiq/switch-api/gsw_flow_pce.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_pce.c b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_pce.c
index 43555faafbb6..c31fcce38c83 100644
--- a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_pce.c
+++ b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_pce.c
@@ -3262,6 +3262,8 @@ int pce_rule_write(void *cdev, ltq_pce_table_t *pthandle, GSW_PCE_rule_t *parm)
 		ptbl.key[12] |= ptable->parse_lsb_idx;
 
 		/* Parser flags 47:32 */
+		ptable->parse1_lsb_idx = 0xFF; //Mark as don't care.
+
 		if (IS_VRSN_31(gswdev->gipver)) {
 			if (parm->pattern.bParserFlag1LSB_Enable == 1) {
 				flag_tbl_t   flags_tbl;
@@ -3276,8 +3278,6 @@ int pce_rule_write(void *cdev, ltq_pce_table_t *pthandle, GSW_PCE_rule_t *parm)
 					return tindex;
 
 				ptable->parse1_lsb_idx = tindex;
-			} else {
-				ptable->parse1_lsb_idx = 0xFF; //Mark as don't care.
 			}
 
 			ptbl.key[16] |= ptable->parse1_lsb_idx;
@@ -3304,6 +3304,8 @@ int pce_rule_write(void *cdev, ltq_pce_table_t *pthandle, GSW_PCE_rule_t *parm)
 		ptbl.key[12] |= (ptable->parse_msb_idx << 8);
 
 		/* Parser flags 63:48 */
+		ptable->parse1_msb_idx = 0xFF; //Mark as don't care.
+
 		if (IS_VRSN_31(gswdev->gipver)) {
 			if (parm->pattern.bParserFlag1MSB_Enable == 1) {
 				flag_tbl_t   flags_tbl;
@@ -3318,8 +3320,6 @@ int pce_rule_write(void *cdev, ltq_pce_table_t *pthandle, GSW_PCE_rule_t *parm)
 					return tindex;
 
 				ptable->parse1_msb_idx = tindex;
-			} else {
-				ptable->parse1_msb_idx = 0xFF; //Mark as don't care.
 			}
 
 			ptbl.key[16] |= (ptable->parse1_msb_idx << 8);
